### DISCUSSION TITLE
使用BMPNAdapter插件，调用adapterOut的时候传了excludeFields属性，但是不起作用，导出的xml中元素上仍然携带了自定义的属性，请问怎么去掉呢

### DISCUSSION THREAD

**Discussion by ycyyf** [didi/LogicFlow#1450] [General]
![image](https://github.com/didi/LogicFlow/assets/38076447/ef48d2c0-197d-4b70-b705-40b4dcd6eb5d)

**Comment by wumail** [comment]
可以参考一下
```typescript
const defaultExcludeFields = {
  in: [],
  out: [
    'properties.panels',
    'properties.nodeSize',
    'properties.definitionId',
    'properties.timerValue',
    'properties.timerType',
    'properties.definitionType',
    'properties.parent',
    'properties.isBoundaryEventTouchingTask',
  ],
};
```

**Reply by ycyyf** [reply]
![image](https://github.com/didi/LogicFlow/assets/38076447/f0bbb3a8-bb4e-4813-8442-d42ae56059df)
![image](https://github.com/didi/LogicFlow/assets/38076447/a143a6d9-48c3-4bc3-9165-aa825ff18760)
还是不行

**Reply by wumail** [reply]
使用方式有一些出入
```typescript
LogicFlow.use(BPMNAdapter, {
   excludeFields: {
    out: ['properties.multiInstanceType']
  }
})
```

**Comment by wumail** [comment]
> ![image](https://private-user-images.githubusercontent.com/38076447/288270885-f0bbb3a8-bb4e-4813-8442-d42ae56059df.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE4MzQ5MDMsIm5iZiI6MTcwMTgzNDYwMywicGF0aCI6Ii8zODA3NjQ0Ny8yODgyNzA4ODUtZjBiYmIzYTgtYmI0ZS00ODEzLTg0NDItZDQyYWU1NjA1OWRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA2VDAzNTAwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWExYjAwYThlOThlN2E3N2JiNDg4MDZmYThiOWZlODcxZjE1YjczMmRhOTI3MzZlMzJhNzlkNjhlOGVmM2NhMDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fCD9kwEviXRDO1d7XmL9bKg9-hZDWbdL6RMhB9pUuAY) ![image](https://private-user-images.githubusercontent.com/38076447/288270920-a143a6d9-48c3-4bc3-9165-aa825ff18760.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE4MzQ5MDMsIm5iZiI6MTcwMTgzNDYwMywicGF0aCI6Ii8zODA3NjQ0Ny8yODgyNzA5MjAtYTE0M2E2ZDktNDhjMy00YmMzLTkxNjUtYWE4MjVmZjE4NzYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA2VDAzNTAwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3MGJlODQ5ZjVmOWRhM2FiMmM1NmUwYjQ3MzE4OGU2M2MzOTRiMTU4MmQ4NmM5ZGY2YzFiNWE5MmEyYjUxOTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-do9vqm3WI-4oUSnnB3G7Ax_0UtpK1yNl4axVOibQ8k) 还是不行

使用方式有一些出入
```typescript
LogicFlow.use(BPMNAdapter, {
   excludeFields: {
    out: ['properties.multiInstanceType']
  }
})
```

**Reply by ycyyf** [reply]
可以了，多谢

**Comment by ycyyf** [comment]
![image](https://github.com/didi/LogicFlow/assets/38076447/c932a5db-a191-4431-9a9e-e120449ea72c)
这个地方怎么改成xsi:type="bpmn:tFormalExpression" 呢，在transform中改不生效
![Uploading image.png…]()

**Reply by ycyyf** [reply]
也是需要放在use里去
LogicFlow.use(BPMNAdapter, {
   excludeFields: {
    out: ['properties.multiInstanceType']
  },
transformer: {
    'bpmn:sequenceFlow': {
      out(data) {
        const {
          properties: { expressionType, condition },
        } = data;
        if (condition) {
          if (expressionType === 'cdata') {
            return {
              json: `<bpmn:conditionExpression xsi:type="bpmn:tFormalExpression"><![CDATA[\${${condition}}]]></bpmn:conditionExpression>`,
            };
          }
          return {
            json: `<bpmn:conditionExpression xsi:type="bpmn:tFormalExpression">${condition}</bpmn:conditionExpression>`,
          };
        }
        return {
          json: '',
        };
      },
    },
  },
})
